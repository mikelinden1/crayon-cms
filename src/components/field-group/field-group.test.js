import FieldGroup from './field-group';

const props = {
    label: 'Test Field',
    name: 'test-field',
    helpText: 'Help!',
    actions: {
        showMediaPicker: () => {}
    }
};

it('FieldGroup matches snapshot', () => {
    const wrapper = shallow(<FieldGroup {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it('FieldGroup shows the correct text', () => {
    const wrapper = shallow(<FieldGroup {...props} />);

    const label = wrapper.find('.control-label').text();
    expect(label).toEqual(props.label);
    
    const helpText = wrapper.find('.help-text').text();
    expect(helpText).toEqual(props.helpText);
});

it('FieldGroup with an error matches snapshot', () => {
    const propsWithErrors = {
        ...props,
        validationErrors: [
            {
                itemName: 'name',
                msg: 'Name is required'
            }
        ]
    };
    const wrapper = shallow(<FieldGroup {...propsWithErrors} />);
    expect(wrapper).toMatchSnapshot();
});

it('FieldGroup with many matches snapshot', () => {
    const propsWithMany = {
        ...props,
        many: true
    };
    const wrapper = shallow(<FieldGroup {...propsWithMany} />);
    expect(wrapper).toMatchSnapshot();
});