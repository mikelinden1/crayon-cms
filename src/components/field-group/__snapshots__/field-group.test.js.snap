// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldGroup matches snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
  >
    Test Field
  </label>
  <Connect(FieldSingle)
    actions={
      Object {
        "showMediaPicker": [Function],
      }
    }
    helpText="Help!"
    label="Test Field"
    name="test-field"
  />
  <p
    className="help-text"
  >
    <small>
      <em>
        Help!
      </em>
    </small>
  </p>
</div>
`;

exports[`FieldGroup with an error matches snapshot 1`] = `
<div
  className="form-group has-error"
>
  <label
    className="control-label"
  >
    Test Field
  </label>
  <Connect(FieldSingle)
    actions={
      Object {
        "showMediaPicker": [Function],
      }
    }
    helpText="Help!"
    label="Test Field"
    name="test-field"
    validationErrors={
      Array [
        Object {
          "itemName": "name",
          "msg": "Name is required",
        },
      ]
    }
  />
  <p
    className="help-text"
  >
    <small>
      <em>
        Help!
      </em>
    </small>
  </p>
  <div
    className="invalid-feedback"
  >
    <p
      key="errormsg-name-0"
    >
      Name is required
    </p>
  </div>
</div>
`;

exports[`FieldGroup with many matches snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
  >
    Test Field
  </label>
  <Connect(FieldMulti)
    actions={
      Object {
        "showMediaPicker": [Function],
      }
    }
    helpText="Help!"
    label="Test Field"
    many={true}
    name="test-field"
  />
  <p
    className="help-text"
  >
    <small>
      <em>
        Help!
      </em>
    </small>
  </p>
</div>
`;
